automation/qbft/runner/scenario.go

```diff
@@ -0,0 +1,50 @@
+package runner
+
+import (
+	"context"
+
+	spectypes "github.com/bloxapp/ssv-spec/types"
+	"go.uber.org/zap"
+
+	"github.com/bloxapp/ssv/ibft/storage"
+	p2pv1 "github.com/bloxapp/ssv/network/p2p"
+	"github.com/bloxapp/ssv/storage/basedb"
+)
+
+// ScenarioFactory creates Scenario instances.
+type ScenarioFactory func(name string) Scenario
+
+// ScenarioContext is the context object that is passed in execution.
+type ScenarioContext struct {
+	Ctx         context.Context
+	LocalNet    *p2pv1.LocalNet
+	Stores      []*storage.QBFTStores
+	KeyManagers []spectypes.KeyManager
+	DBs         []basedb.IDb
+}
+
+// Bootstrapper bootstraps the given scenario.
+type Bootstrapper func(ctx context.Context, logger *zap.Logger, scenario Scenario) (*ScenarioContext, error)
+
+type scenarioCfg interface {
```

Comment:
We should define something more verbose to allow us to specify the cases, these configs are just a subset of possible networking parameters.
Besides that we should define things like a slice of duties to execute and the expected states in each node